### PR TITLE
[DEV-7509] Number of New Awards

### PR DESCRIPTION
**Pull request by sethstoudenmier**
**Description:**
Endpoint that returns the number of new awards for an Agency and Fiscal Year.

**Technical details:**
Made adjustments to `query_with_filters.py` to allow for re-use in this endpoint when filtering down the Awards that should be counted.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [x] Matview impact assessment completed
5. [x] Frontend impact assessment completed
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-7509](https://federal-spending-transparency.atlassian.net/browse/DEV-7509):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected (API | Script | Download)
    - [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/api_contracts/contracts/v2/agency/toptier_code/awards/new/count.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        + `fiscal_year` (optional, number)
+            The desired appropriations fiscal year. Defaults to the current FY
+        + `agency_type` (optional, enum[string])
+            The agency type to pull the overview for.
+            + Default: `awarding`
+            + Members
+                + `awarding`
+                + `funding`
+        + `award_type_codes` (optional, array[AwardTypeCodes], fixed-type)
+            + Default: `["02", "03", "04", "05", "07", "08", "10", "06", "09", "11", "A", "B", "C", "D", "IDV_A", "IDV_B", "IDV_B_A", "IDV_B_B", "IDV_B_C", "IDV_C", "IDV_D", "IDV_E"]`
```
Comment (collinwr) [inline usaspending_api/api_contracts/contracts/v2/agency/toptier_code/awards/new/count.md:29]: This default example seems a little tricky. The API will not accept quotes or spaces in this list

Comment (emily-brents) [reply usaspending_api/api_contracts/contracts/v2/agency/toptier_code/awards/new/count.md:29]: You also don't need to provide all the type codes as a default - if you provide a blank/null for the award type code filter, the filter won't get added to the elasticsearch query and it'll work as if you are filtering by all type codes

Comment (sethstoudenmier) [reply usaspending_api/api_contracts/contracts/v2/agency/toptier_code/awards/new/count.md:29]: Fine with removing it for me. Dredd keeps complaining about the default anyways.

File: usaspending_api/common/query_with_filters.py [hunk 2]
```diff
@@ -525,7 +493,9 @@ class QueryWithFilters:
     unsupported_filters = ["legal_entities"]
 
     @classmethod
-    def _generate_elasticsearch_query(cls, filters: dict, query_type: _QueryType, nested_path: str = "") -> ES_Q:
+    def _generate_elasticsearch_query(cls, filters: dict, query_type: _QueryType, **options) -> ES_Q:
+        nested_path = options.pop("nested_path", "")
```
Comment (collinwr) [inline usaspending_api/common/query_with_filters.py:497]: Why is the option retrieved with a `pop` here, whereas the other options use a `get`?

File: usaspending_api/search/filters/elasticsearch/tas.py [hunk 3]
```diff
@@ -11,7 +11,7 @@ class TasCodes(_Filter, HierarchicalFilter):
     underscore_name = "tas_codes"
 
     @classmethod
-    def generate_elasticsearch_query(cls, filter_values, query_type: _QueryType, nested_path: str = "") -> ES_Q:
+    def generate_elasticsearch_query(cls, filter_values, query_type: _QueryType, **options) -> ES_Q:
         if isinstance(filter_values, list):
             # This is a legacy usage, and will be dealt with by the other filter
             return TreasuryAccounts.generate_elasticsearch_query(filter_values, query_type)
```
Comment (collinwr) [inline usaspending_api/search/filters/elasticsearch/tas.py:17]: Why not give the `TreasuryAccounts.generate_elasticsearch_query(filter_values, query_type)` method an `options` parameter?

Comment (sethstoudenmier) [reply usaspending_api/search/filters/elasticsearch/tas.py:17]: Simply overlooked it when updating all methods that implemented `_Filter`. Will get that changed.

**Review by collinwr — DISMISSED** [review]

**Review by afnazu — COMMENTED** [review]
API contract approved

**Review by collinwr — APPROVED** [review]

**Pull request merged**